src/utils/particle/bubbler.ts

```diff
@@ -25,19 +25,23 @@ export class Bubbler {
     public bubble() {
         const container = this.container;
         const options = container.options;
+        const hoverEnabled = options.interactivity.events.onhover.enable;
+        const hoverMode = options.interactivity.events.onhover.mode;
+        const clickEnabled = options.interactivity.events.onclick.enable;
+        const clickMode = options.interactivity.events.onclick.mode;
 
         /* on hover event */
-        if (options.interactivity.events.onhover.enable && Utils.isInArray(HoverMode.bubble, options.interactivity.events.onhover.mode)) {
+        if (hoverEnabled && Utils.isInArray(HoverMode.bubble, hoverMode)) {
             this.hoverBubble();
-        } else if (options.interactivity.events.onclick.enable && Utils.isInArray(ClickMode.bubble, options.interactivity.events.onclick.mode)) {
+        } else if (clickEnabled && Utils.isInArray(ClickMode.bubble, clickMode)) {
             this.clickBubble();
         }
     }
 
-    public process(dist_mouse: number, time_spent: number, bubble_param: number, particles_param: number, p_obj_bubble: number | undefined, p_obj: number, id: ProcessBubbleType) {
+    private process(dist_mouse: number, time_spent: number, bubble_param: number, particles_param: number, p_obj_bubble: number | undefined, p_obj: number, id: ProcessBubbleType) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'process' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5011230)